From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Boris Krasnovskiy <boris.krasnovskiy@ezurio.com>
Date: Thu, 8 Oct 2026 00:00:00 +0000
Subject: [PATCH] mt8188: load BL32 without a header

Summit boots the headerless tee-pager_v2.bin, so BL2 loads BL32 at BL32_BASE
instead of BL32_BASE - BL32_HEADER_SIZE.

Upstream-Status: Inappropriate [Summit SoM specific]

Signed-off-by: Boris Krasnovskiy <boris.krasnovskiy@ezurio.com>
---
diff --git a/plat/mediatek/mt8188/bl2_plat_setup.c b/plat/mediatek/mt8188/bl2_plat_setup.c
--- a/plat/mediatek/mt8188/bl2_plat_setup.c
+++ b/plat/mediatek/mt8188/bl2_plat_setup.c
@@ -90,7 +90,7 @@
 
 		SET_STATIC_PARAM_HEAD(image_info, PARAM_EP,
 			VERSION_2, image_info_t, 0),
-		.image_info.image_base = BL32_BASE - BL32_HEADER_SIZE,
+		.image_info.image_base = BL32_BASE,
 		.image_info.image_max_size = BL32_LIMIT,
 
 		.next_handoff_image_id = BL33_IMAGE_ID,
